tests/ping.go

```diff
@@ -0,0 +1,103 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package tests
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	expect "github.com/google/goexpect"
+	"google.golang.org/grpc/codes"
+	"k8s.io/utils/net"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/client-go/log"
+)
+
+const promptExpression = `(\$ |\# )`
+const termNLExpression = "\r\n"
+
+var (
+	shellSuccess = regexp.MustCompile("\n0" + termNLExpression + ".*" + promptExpression)
+	shellFail    = regexp.MustCompile("\n[1-9].*" + termNLExpression + ".*" + promptExpression)
```

Comment:
you can use `tests.RetValue("[1-9]")

Reply:
error code can be 2 / 3 digits, for example gnu ping return 255 
even if it will work because of the `.*` that will match the rest, it would be implicit.
as i wrote before also, i dont see a reason to split between 0 and the rest explicity here, then just use EXP \n0 as we do on all the other places.
only benefit is that it fail fast without using the timeout, but just one place doesnt worth it, 
unless this whole case would be a util we use on all the places.

Reply:
> error code can be 2 / 3 digits, for example gnu ping return 255
> even if it will work because of the `.*` that will match the rest, it would be implicit.

What do you mean "even if it will work"? It works period. Any string that does not start with "0" is matched.
No idea what is implicit about it.
 
> as i wrote before also, i dont see a reason to split between 0 and the rest explicity here, then just use EXP \n0 as we do on all the other places.
> only benefit is that it fail fast without using the timeout, but just one place doesnt worth it,
> unless this whole case would be a util we use on all the places.

The benefit is exactly what you mentioned, avoiding the timeout.
In the past it was 30sec timeout, not it is 10sec timeout. I see no reason to pay for it if it can be avoided.
I think ping is frequently used and making a better and faster one is a good step forward.

Reply:
> > error code can be 2 / 3 digits, for example gnu ping return 255
> > even if it will work because of the `.*` that will match the rest, it would be implicit.
> 
> What do you mean "even if it will work"? It works period. Any string that does not start with "0" is matched.
> No idea what is implicit about it.

i know it works - i just say that it works because the `.*` match the other digits if exists, but its implicit that there can be couple digits there,
because we see in our eyes just 1 digit at first glance.

> 
> > as i wrote before also, i dont see a reason to split between 0 and the rest explicity here, then just use EXP \n0 as we do on all the other places.
> > only benefit is that it fail fast without using the timeout, but just one place doesnt worth it,
> > unless this whole case would be a util we use on all the places.
> 
> The benefit is exactly what you mentioned, avoiding the timeout.
> In the past it was 30sec timeout, now it is 10sec timeout. I see no reason to pay for it if it can be avoided.
> I think ping is frequently used and making a better and faster one is a good step forward.

so as written in the last sentence, "unless this whole case would be a util we use on all the places."
might worth a follow PR ?

Reply:
> > What do you mean "even if it will work"? It works period. Any string that does not start with "0" is matched.
> > No idea what is implicit about it.
> 
> i know it works - i just say that it works because the `.*` match the other digits if exists, but its implicit that there can be couple digits there,
> because we see in our eyes just 1 digit at first glance.

I see with my eyes a regex that says `[1-9].*` which means it starts with a digit that is not 0 and continues with any number of characters after.

> > The benefit is exactly what you mentioned, avoiding the timeout.
> > In the past it was 30sec timeout, now it is 10sec timeout. I see no reason to pay for it if it can be avoided.
> > I think ping is frequently used and making a better and faster one is a good step forward.
> 
> so as written in the last sentence, "unless this whole case would be a util we use on all the places."
> might worth a follow PR ?

If I or someone else will want to use this again, this can be promoted to a generalized helper. I suspect it will happen, but I do not know at this moment.